pkg/3scale/amp/operator/redis_reconciler.go

```diff
@@ -19,6 +22,95 @@ func NewRedisReconciler(baseAPIManagerLogicReconciler *BaseAPIManagerLogicReconc
 	}
 }
 
+// RedisDependencyReconciler is a generic DependencyReconciler that reconciles
+// an internal Redis instance using the Redis options
+type RedisDependencyReconciler struct {
```

Comment:
The existing `RedisReconciler` is not being used. Maybe remove that unused one and use that name for this struct?